experimental/rsocket-src/RSocketConnectionHandler.cpp

```diff
@@ -22,20 +23,19 @@ class RSocketHandlerBridge : public reactivesocket::DefaultRequestHandler {
   void handleRequestStream(
       Payload request,
       StreamId streamId,
-      const std::shared_ptr<Subscriber<Payload>>& response) noexcept override {
+      const yarpl::Reference<yarpl::flowable::Subscriber<Payload>>& response) noexcept override {
     auto flowable =
         handler_->handleRequestStream(std::move(request), std::move(streamId));
     // bridge from the existing eager RequestHandler and old Subscriber type
     // to the lazy Flowable and new Subscriber type
-    flowable->subscribe(yarpl::Reference<yarpl::flowable::Subscriber<Payload>>(
-        new NewToOldSubscriber(std::move(response))));
+    flowable->subscribe(std::move(response));
   }
 
-  std::shared_ptr<Subscriber<Payload>> handleRequestChannel(
+  yarpl::Reference<yarpl::flowable::Subscriber<Payload>> handleRequestChannel(
       Payload request,
       StreamId streamId,
-      const std::shared_ptr<Subscriber<Payload>>& response) noexcept override {
-    auto eagerSubscriber = std::make_shared<EagerSubscriberBridge>();
+      const yarpl::Reference<yarpl::flowable::Subscriber<Payload>>& response) noexcept override {
+    auto eagerSubscriber = make_ref<EagerSubscriberBridge>();
```

Comment:
Discussed ... ignoring this for now. It's the hard-to-understand inverted API since it's eager.